frontend/tests/unit/components/note/NoteCard.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { createTestingPinia } from '@pinia/testing';
+import { shallowMount } from '@vue/test-utils';
+
+import NoteCard from '@/components/note/NoteCard.vue';
+import { userService } from '@/services';
+import { StorageKey } from '@/utils/constants';
+import PrimeVue from 'primevue/config';
+
+import type { AxiosResponse } from 'axios';
+
+const useUserService = vi.spyOn(userService, 'searchUsers');
+
+vi.mock('vue-router', () => ({
+  useRouter: () => ({
+    push: vi.fn(),
+    replace: vi.fn()
+  })
+}));
+
+const emptyTestPinia = () => {
+  return createTestingPinia({
```

Comment:
Optional Style: As this function is a single function invocation and return, you could inline this into a one liner embed and avoid needing to use an explicit body and return statement:
```ts
const emptyTestPinia = () => createTestingPinia({
// the rest of argument
```